File: ServidorUDP/Python/udp_server.py
import socket

HOST = "0.0.0.0"
PORT = 5000

sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
sock.bind((HOST, PORT))

print("Servidor UDP escuchando en puerto", PORT)

while True:
    data, addr = sock.recvfrom(1024)
    print("Mensaje recibido:", data.decode(), "de", addr)

    sock.sendto(data, addr)

File: ServidorUDP/Java/UdpServer.java
package com.socket.udp;

import java.net.DatagramPacket;
import java.net.DatagramSocket;

public class UdpServer {
    public static void main(String[] args) throws Exception {

        DatagramSocket socket = new DatagramSocket(5001);
        byte[] buffer = new byte[1024];

        System.out.println("Servidor UDP escuchando en puerto 5001");

        while (true) {

            DatagramPacket packet = new DatagramPacket(buffer, buffer.length);

            socket.receive(packet);

            String mensaje = new String(packet.getData(), 0, packet.getLength());

            System.out.println("Mensaje recibido: " + mensaje);
            System.out.println("Address: "+packet.getAddress());
            System.out.println("Port: "+packet.getPort());
            DatagramPacket response = new DatagramPacket(
                    packet.getData(),
                    packet.getLength(),
                    packet.getAddress(),
                    packet.getPort()
            );

            socket.send(response);
        }
    }
}
